properties/test_netcdf_roundtrip.py

```diff
@@ -0,0 +1,49 @@
+"""
+Property-based tests for round-tripping data to netCDF
+"""
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+from hypothesis import given, settings
+
+import xarray as xr
+
+# Run for a while - arrays are a bigger search space than usual
+settings.register_profile("ci", deadline=None)
+settings.load_profile("ci")
+
+
+an_array = npst.arrays(
+    dtype=st.one_of(
+        npst.unsigned_integer_dtypes(),
+        npst.integer_dtypes(),
+        # NetCDF does not support float16
+        # https://www.unidata.ucar.edu/software/netcdf/docs/data_type.html
+        npst.floating_dtypes(sizes=(32, 64)),
```

Comment:
I think we could add `byte_string_dtypes`, `unicode_string_dtypes`, `datetime64_dtypes` and `timedelta64_dtypes` to this list.